src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java

```diff
@@ -352,20 +353,26 @@ public void deleteNotifications(ProgrammeMembership programmeMembership)
   }
 
   /**
-   * Helper function to determine whether a notification should be scheduled.
+   * Helper function to determine whether a programme created notification should be scheduled.
    *
-   * @param milestone                The milestone to consider.
+   * @param programmeMembership      The updated programme membership to consider.
    * @param notificationsAlreadySent The notifications already sent for this entity.
    * @return true if it should be scheduled, false otherwise.
    */
-  private boolean shouldScheduleNotification(NotificationType milestone,
-      Map<NotificationType, Instant> notificationsAlreadySent) {
+  private boolean shouldScheduleProgrammeCreatedNotification(
+      ProgrammeMembership programmeMembership,
+      Map<NotificationType, History> notificationsAlreadySent) {
 
-    //do not resend any notification
-    if (notificationsAlreadySent.containsKey(milestone)) {
+    //only resend deferred programme notifications
+    if (notificationsAlreadySent.containsKey(PROGRAMME_CREATED)) {
       return false;
+//      TODO: to be implemented with the actual scheduling in next PR
```

Comment:
SonarCloud is not a fan of all this, but I've left it in to show the direction of travel.